src/lib/core/CHIPTLVText.cpp

```diff
@@ -0,0 +1,231 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2015-2017 Nest Labs, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <core/CHIPTLVText.hpp>
+#include <core/CHIPTLVUtilities.hpp>
+#include <support/CodeUtils.h>
+#include <core/CHIPTLVTypes.h>
+#include <cstdint>
+
+namespace chip {
+
+namespace TLV {
+
+namespace Debug {
+
+static CHIP_ERROR Print(TLV::TLVReader &reader, PrinterFunc printerFunc, int tabLevel, bool isFirst);
+
+CHIP_ERROR Print(TLV::TLVReader &reader, PrinterFunc printerFunc) {
+    return Print(reader, printerFunc, 0, true);
+}
+
+void PrintTabs(PrinterFunc printerFunc, int tabLevel) 
+{
+    for (int i = 0; i < tabLevel; i++) {
+        printerFunc("\t");
```

Comment:
Given how inconsistent tab stop handling is (e.g. quoting a block of text using tabs might wreak havoc on it), it might be better to just use `"  "` or `"    "` here.